### ISSUE TITLE
Disallow 'declare' modifier on private named properties

### ISSUE THREAD

**Issue by ajafff** [microsoft/TypeScript#36345]
**TypeScript Version:** 3.8.0-dev.20200119

**Search Terms:**

**Code**

```ts
class C {
    declare #foo: number;
    constructor() {
        this.#foo = 1;
    }
}
```

**Expected behavior:**

Error on `declare` modifier as it doesn't make sense for private named properties.

**Actual behavior:**

No compile error. Instead the emit is invalid. Depending on the target either the property declaration is missing (ESNext) or the whole downleveling is not done at all.

**Playground Link:** http://www.typescriptlang.org/play/?target=7&ts=3.8.0-dev.20200119#code/MYGwhgzhAEDC0G8BQ1XQCYFNRgE6egGIAzAe1IC5oA7AVwFsAjTXAbhTWFOogBddawXqVwAKAJSIOaNLwAWASwgA6EuWgBeaAEZ2MgL5J9QA

**Related Issues:**

**Comment by DanielRosenwasser** [comment]
Same with `abstract` too.

**Comment by ajafff** [comment]
`abstract` is already an error, there's even a test for that. Fix for `declare` is up at #36381
